MapboxVision/Services/LocationManager.swift

```diff
@@ -41,11 +41,12 @@ final class LocationManager: NSObject, CLLocationManagerDelegate {
     }
 
     func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
-        guard let handler = locationHandler else { return }
+        guard let handler = locationHandler, isStarted else { return }
         locations.forEach(handler)
     }
 
     func locationManager(_ manager: CLLocationManager, didUpdateHeading newHeading: CLHeading) {
+        guard isStarted else { return }
```

Comment:
We do not check the `headingHandler` for being `nil` here as for `locationManager:didUpdateLocations:` method - might be a bit confusing IMHO.

Or we can leave only one check for `isStarted` above (please take a look at related comment above).

Reply:
The guard in `didUpdateLocation` is there to unwrap handler before passing it to foreach. There's no need for it here.

Reply:
Thanks for the explanation!